File: Watcher.BLL/Services/ITestService.cs
using Watcher.BLL.Models;
using Watcher.BLL.Models.Tests;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Watcher.BLL.Services
{
	public interface ITestService : ICrudService<Test, AddTestData>
	{
		Task<DefaultDataFetchResult<ICollection<Test>>> GetAllAsync(int userId);
		Task<DefaultFetchResult> AddAsync(AddTestData entity, int userId);
		Task<DefaultFetchResult> UpdateAsync(Test entity, int userId);
		Task<DefaultFetchResult> DeleteAsync(int id, int userId);

		Task InitAsync();
	}
}

File: client_app/packages/watcher_client_bll/lib/services/test/itest_service.dart
import 'package:watcher_client_bll/models/test/add_test.dart';
import 'package:watcher_client_bll/models/test/test.dart';
import 'package:watcher_client_bll/services/icrud_service.dart';

abstract class ITestService extends ICrudService<Test, AddTest> {}